app/modules/visualization/js/objectMapLeflet.js

```diff
@@ -160,29 +106,53 @@ policycompass.viz.mapLeaflet = function (options) {
             var initialLng = self.initialLng;
         }
     }
-    //console.log(initialLat);
-    //console.log(initialLng);
-
-    var map = L.map("mapPC_" + self.idName, {
-        layers: [
-            L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
-                //attribution: 'Policy Compass &copy;',
-                noWrap: true
-            })
-        ],
-        zoom: initialZoom,
-        center: [initialLat, initialLng], //center: [49.009952, 0],
-        minZoom: 1,
-        maxZoom: 8,
-        maxBounds: [[-90.0, -180.0], [90.0, 180.0]]
-        //maxBounds: [[-85.0, -180.0],[85.0, 180.0]]
-
-    });
-
-    L.control.pan().addTo(map);
-
-    L.control.scale().addTo(map);
 
+	//remove info divs
+	var elements = document.getElementsByClassName("info_"+self.idName);
+	while(elements.length > 0) {
+       	elements[0].parentNode.removeChild(elements[0]);
+    }
+    //remove legend divs
+	var elements = document.getElementsByClassName("legend_"+self.idName);
+	while(elements.length > 0) {
+       	elements[0].parentNode.removeChild(elements[0]);
+    }
+    //remove elements divs
+	var elements = document.getElementsByClassName("leaflet-clickable");
+	while(elements.length > 0) {
+       	elements[0].parentNode.removeChild(elements[0]);
+    }
+    
+	//used to check if map exist
+	if (ng_el_scope.map != undefined) {
+		ng_el_scope.reload = true;
+		var map = ng_el_scope.map; 
+	}
+	else {	    
+	    var map = L.map("mapPC_" + self.idName, {
+	        layers: [
+	            L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {

```

Comment:
shouldn't this be 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png' since we are using SSL? I would also recommend to make this configurable via app configuration